File: app/src/main/java/com/example/kent/hyperdeals/NavigationOptionsActivity/NotificationReceiver.java
package com.example.kent.hyperdeals.NavigationOptionsActivity;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

public class NotificationReceiver extends BroadcastReceiver {
    String TAG = "Receiver";
    @Override
    public void onReceive(Context context, Intent intent) {

        if (intent.getAction().equalsIgnoreCase("INTERESTED")) {
            Log.e(TAG,"INTERESTED");
//            Toast.makeText(context, "Booking your ride", Toast.LENGTH_SHORT).show();


        } else if (intent.getAction().equalsIgnoreCase("DISMISS")) {
//
            Log.e(TAG,"DISMISS");

//            NotificationManager notificationManager =
//                    (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
//            notificationManager.cancel(11111);

        }



    }
}

File: app/src/main/java/com/example/kent/hyperdeals/NotificationReceiver.kt
package com.example.kent.hyperdeals
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.support.v4.app.NotificationManagerCompat
import android.util.Log
import android.widget.Toast
import com.example.kent.hyperdeals.Model.PromoModel

class NotificationReceiver: BroadcastReceiver() {
  var TAG = "NotificationReceiver"

    override fun onReceive(context: Context, intent: Intent) {
        var myPromo =   intent.getParcelableExtra<PromoModel>("message")
        var myNotificationID = intent.getIntExtra("notificationID",0)
        Log.e(TAG, "NotificationReceiver ${myPromo.promoID}  ${myPromo.promoname} $myNotificationID")
        val notificationManagerCompat = NotificationManagerCompat.from(context)
        notificationManagerCompat.cancel(myNotificationID)



    }
}
class NotificationReceiverDismiss : BroadcastReceiver() {
    var TAG = "NotificationReceiver"

    override fun onReceive(context: Context, intent: Intent) {
        var myPromo =   intent.getParcelableExtra<PromoModel>("message")
        var myNotificationID = intent.getIntExtra("notificationID",0)
        Log.e(TAG, "NotificationReceiver ${myPromo.promoID}  ${myPromo.promoname} $myNotificationID")
        val notificationManagerCompat = NotificationManagerCompat.from(context)
        notificationManagerCompat.cancel(myNotificationID)


    }
}

